Yoast/YoastSEO.js
# Adds passive voice assessment for French

## Summary

This PR can be summarized in the following changelog entry:

Adds a passive voice assessment for French.

## Relevant technical choices:

* Moves functionality from the English passive voice assessement to a shared `passivevoice` directory.

## Test instructions

This PR can be tested by following these steps:

- Check out the branch.
- In `src/assessments/readability/passiveVoiceAssessment.js` add `"fr"` to the available languages.
- Use the browserified example. Don't forget to run a `grunt build:js`.
-  Set the locale to `fr_FR`.
- Make sure the following sentences are marked as passive:
  - `J'étais mangé.` (passive sentence with regular participle)
  - `J’ai été mangé.` (passive sentence with the auxiliary `été`)
  - `Il était vendu.` (passive sentence with irregular participle)
  - `Elle était vendue.` (passive sentence with an irregular passive plus suffix)
  - `Il était mû.` (passive sentence with an irregular participle with irregular conjugation pattern)
  - `Une réponse a été promise à maintes reprises.` (passive sentence with an irregular participle ending in -s)
  - `Le jury de thèse de Mme Randerson était-il informé de cela ?` (passive sentence with an inverted auxiliary in a question)

- Make sure the following sentences are **not** marked as passive:
  - `Je mange.` (active sentence without participle)
  - `J'ai mangé.` (active sentence with regular participle)
  - `J'ai vendu.` (active sentence with irregular participle)
  - `J'étais allé.` (active sentence with passive auxiliary and non-passive participle from adjective/verb exception list)
  - `Elle est la sœur aînée.` (active sentence with passive auxiliary and non-passive é-word from adjective/verb exception list plus suffix)
  - `J’étais au café.` (active sentence with passive auxiliary and non-passive é-word from noun exception list)
  - `Ce sont deux cafés sympas.`  (active sentence with passive auxiliary and non-passive é-word from noun exception list plus suffix)
  - `Est-ce que la création de cet outil contribuera à améliorer l’intégrité scientifique ?` (active sentence with with passive auxiliary and non-passive é-word from noun exception list plus contracted article)
  - `Elle est triste malgré tout.`   (active sentence with passive auxiliary and non-passive é-word from "other" exception list)
  - `Ils se sont lavés.` (active sentence with être-auxiliary that's preceded by a reflexive pronoun)
  - `Cela est le film le plus vu.` (active sentence with être-auxiliary that's followed by an article)
  - `Elle est en vue.` (active sentence with a participle that's marked as non-passive by a direct precedence exception)
  - `Il n’est pas nécessaire d’avoir vu le premier film.` (active sentence with a participle that's marked as non-passive by a precedence exception)
  - `Il s'est lavé.` (active sentence with être-auxiliary that's fused with a reflexive pronoun).

@agnieszkaszuba is going to test this assessment more extensively with a wider range of texts before it can be merged.

Fixes #1286
